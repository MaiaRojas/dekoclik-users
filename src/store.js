import { createStore, applyMiddleware, compose } from 'redux';
import { reactReduxFirebase } from 'react-redux-firebase';
import { reduxFirestore } from 'redux-firestore';
import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import reducers from './reducers';
import apiMiddleware from './middleware';


firebase.initializeApp({
  apiKey: process.env.FIREBASE_API_KEY,
  authDomain: `${process.env.FIREBASE_PROJECT}.firebaseapp.com`,
  databaseURL: `https://${process.env.FIREBASE_PROJECT}.firebaseio.com`,
  projectId: process.env.FIREBASE_PROJECT,
  storageBucket: `${process.env.FIREBASE_PROJECT}.appspot.com`,
  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
});


const firestore = firebase.firestore();
firestore.settings({ timestampsInSnapshots: true });


const middlewares = [apiMiddleware];


// Add redux Firebase to compose
const createStoreWithFirebase = compose(
  reactReduxFirebase(firebase, {
    userProfile: 'users',
    useFirestoreForProfile: true,
  }),
  reduxFirestore(firebase),
)(createStore);


// if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  // const { logger } = require('redux-logger');
  // middlewares.push(logger);
// }


export default createStoreWithFirebase(reducers, applyMiddleware(...middlewares));
