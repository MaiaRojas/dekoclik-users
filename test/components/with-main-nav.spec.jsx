import React from 'react';
import { shallow, render } from 'enzyme';
import { Provider } from 'react-redux';
import Intl from '../../src/intl';
import store from '../../src/store';
import WithMainNav from '../../src/components/with-main-nav';


describe('<WithMainNav />', () => {

  it('should print warning when missing component prop', () => {
    const spy = jest.spyOn(console, 'error').mockImplementation(jest.fn());
    shallow(<WithMainNav />);

    expect(spy.mock.calls.length > 0).toBe(true);
    expect(spy.mock.calls[0]).toMatchSnapshot();

    spy.mockReset();
    spy.mockRestore();
  });

  it.skip('should spread props to Component', () => {
    const Component = props => (
      <div className="my-component">
        hello {props.auth.displayName} (path is {props.match.path})
      </div>
    );
    const wrapper = render((
      <Provider store={store}>
        <Intl profile={{ locale: 'es-ES' }}>
          <WithMainNav
            component={Component}
            auth={{ displayName: 'Ada Lovelace', email: 'ada@gmail.com' }}
            match={{ path: '/' }}
            history={{ push: () => {} }}
            firebase={{ logout: () => {} }}
          />
        </Intl>
      </Provider>
    ));

    console.log(wrapper);
    // const children = wrapper.children();
    // expect(children.length).toEqual(1);
    // console.log(children.children()[0]);
    // expect(children[1].children[0].attribs).toMatchSnapshot();
    // expect(children[1].children[0].children[0].type).toBe('text');
    // expect(children[1].children[0].children[0].data).toBe('hello Ada Lovelace (path is /)');
  });

  it.skip('should wrap component in container along with MainNav', () => {
    const Component = () => (<div className="my-component">hello world</div>);

    const wrapper = render((
      <Provider store={store}>
        <WithMainNav
          component={Component}
          auth={{ displayName: 'Ada Lovelace', email: 'ada@gmail.com' }}
          match={{ path: '/' }}
          history={{ push: () => {} }}
          firebase={{ logout: () => {} }}
        />
      </Provider>
    ));

    expect(wrapper.length).toBe(1);
    expect(wrapper[0].type).toBe('tag');
    expect(wrapper[0].name).toBe('div');
    expect(wrapper[0].attribs.class).toMatch(/^app /);
    expect(wrapper[0].children.length).toBe(2);

    const $muiHiddenCss = wrapper[0].children[0].children[0];
    expect($muiHiddenCss.attribs.class).toMatch(/MuiHiddenCss-smDown-\d+/);

    const $drawer = $muiHiddenCss.children[0];
    expect($drawer.attribs.class).toMatch(/MuiDrawer-docked-\d+/);
    expect($drawer.attribs.class).toMatch(/LeftDrawer-drawer-\d+/);

    expect(wrapper.find('.my-component').length).toBe(1);
  });

});
