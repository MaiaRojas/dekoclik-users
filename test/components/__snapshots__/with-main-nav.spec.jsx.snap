// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithMainNav /> should print warning when missing component prop 1`] = `
Array [
  "Warning: Failed prop type: The prop \`component\` is marked as required in \`WithMainNav\`, but its value is \`undefined\`.
    in WithMainNav",
]
`;

exports[`<WithMainNav /> should spread props to Component 1`] = `
Object {
  "class": "my-component",
}
`;
